src/pages/Chat/components/messages/ImageGallery.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect, type FC } from "react";
+import { ImageList, ImageListItem } from "@mui/material";
+
+type Props = {
+  attachments: string[];
+};
+
+type ImageSize = {
+  width: number;
+  height: number;
+};
+
+const ImageGallery: FC<Props> = ({ attachments }) => {
+  const [images, setImages] = useState<
+    { url: string; rows: number; cols: number }[]
+  >([]);
+
+  const getImageSize = (url: string): Promise<ImageSize> => {
+    return new Promise((resolve) => {
+      const img = new Image();
+      img.src = url;
+      img.onload = () => resolve({ width: img.width, height: img.height });
+      img.onerror = () => resolve({ width: 1, height: 1 });
+    });
+  };
+
+  const calculateImageSize = (size: ImageSize) => {
+    const aspectRatio = size.width / size.height;
+
+    const cols = aspectRatio > 1.5 ? 1 : 2;
```

Comment:
can be  written  in shorter way  return { rows:aspectRatio > 1.5 ? 1 : 2 :, cols:aspectRatio > 1.5 ? 1 : 2 };